h/views/activity.py

```diff
@@ -162,6 +162,27 @@ def user_annotation_count(aggregation, userid):
         return result
 
     @view_config(request_method='POST',
+                 request_param='group_join')
+    def join(self):
+        """
+        Join the given group.
+
+        This adds the authenticated user to the given group and redirect the
+        browser to the search page.
+        """
+        if not self.request.has_permission('join', self.group):
```

Comment:
Yep, I added it explicitly to make sure it's harder to introduce a bug where a user could join a non-joinable group.